src/os_posix/os_union_fs.c

```diff
@@ -1006,21 +1009,32 @@ __union_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *nam
             WT_ERR(__union_fs_open_in_source(union_fs, session, union_fh, flags));
 
             if (!dest_exist) {
-                // We're creating a new destination file which is backed by a source file. It
-                // currently has a length of zero, but we want its length to be the same as the
-                // source file. We do this by reading the last byte of the file. This will read
-                // promote the byte into the destination file, setting the file size.
+                /*
+                 * We're creating a new destination file which is backed by a source file. It
+                 * currently has a length of zero, but we want its length to be the same as the
+                 * source file. We do this by reading the last byte of the file. This will read
```

Comment:
This comment seems out of date wrt the truncate logic.

Reply:
Removed: [9f8e50e](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/9f8e50e9e9e25c96dd296661234032fd94c5c45c)